### PR TITLE
fix: fix sp exit

### PR DESCRIPTION
**Pull request by alexgao001**
### Description

1. fix the offset while iterating objects in GVG.
2. Union recover pieces
3. whitelist recovers data from secondary SP as successor secondary

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### Potential Impacts
* add potential impacts for other components here
* ...

### CODE REVIEW

File: modular/gater/admin_handler.go [hunk 1]
```diff
@@ -877,27 +877,42 @@ func (g *GateModular) getRecoverPiece(ctx context.Context, objectInfo *storagety
...
-		if err != nil {
-			return nil, ErrConsensusWithDetail("query swapInInfo err: " + err.Error())
-		}
-		successorSP, err = g.baseApp.Consensus().QuerySPByID(ctx, swapInInfo.SuccessorSpId)
-		if err != nil {
-			return nil, ErrConsensusWithDetail("query sp err: " + err.Error())
-		}
-		if primarySp.Id == swapInInfo.TargetSpId && successorSP.OperatorAddress == signatureAddr.String() {
-			isSuccessor = true
+		swapInInfo, err = g.baseApp.Consensus().QuerySwapInInfo(ctx, gvg.FamilyId, virtualgrouptypes.NoSpecifiedGVGId)
```
Comment (ruojunm) [inline modular/gater/admin_handler.go:887]: Need we update test file to cover and verify this long section of changes?

File: modular/manager/recover_scheduler.go [hunk 2]
```diff
@@ -28,8 +28,8 @@ const (
 	MaxRecoveryTime  = 50
 
 	recoverInterval     = 30 * time.Second
-	verifyInterval      = 1 * time.Minute
-	verifyGVGQueryLimit = uint32(50)
+	verifyInterval      = 10 * time.Second
+	verifyGVGQueryLimit = uint32(100)
```
Comment (ruojunm) [inline modular/manager/recover_scheduler.go:32]: should this be committed to develop branch?

File: modular/gater/admin_handler.go (hunk 1 above)
Comment (alexgao001) [reply modular/gater/admin_handler.go:887]: make it compatible for more case, so the test wont break.

**Review by constwz — APPROVED** [review]

File: modular/manager/recover_scheduler.go (hunk 2 above)
Comment (alexgao001) [reply modular/manager/recover_scheduler.go:32]: Yes

**Review by clydemeng — APPROVED** [review]

**Review by ruojunm — APPROVED** [review]

**Pull request merged**